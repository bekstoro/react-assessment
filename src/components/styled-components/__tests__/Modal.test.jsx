import React from 'react';
import { render } from '@testing-library/react';
import 'jest-styled-components';

import {
  BaseModal,
  BigModal,
  LoadingIcon,
  LoadingText,
  ModalButtons,
  ModalCancellationButton,
  ModalConfirmationButton,
  ModalContent,
  ModalError,
  ModalInput,
  ModalText,
  ModalTitle,
  RedIcon,
  SmallModal,
  SmallModalCentered,
} from '@components/styled-components/Modal';

describe('Modal styled components', () => {
  it('BaseModal renders correctly', () => {
    const { container } = render(<BaseModal />);
    expect(container).toMatchSnapshot();
  });

  it('BigModal renders correctly', () => {
    const { container } = render(<BigModal />);
    expect(container).toMatchSnapshot();
  });

  it('SmallModal renders correctly', () => {
    const { container } = render(<SmallModal />);
    expect(container).toMatchSnapshot();
  });

  it('SmallModalCentered renders correctly', () => {
    const { container } = render(<SmallModalCentered />);
    expect(container).toMatchSnapshot();
  });

  it('LoadingIcon renders correctly', () => {
    const { container } = render(<LoadingIcon />);
    expect(container).toMatchSnapshot();
  });

  it('LoadingText renders correctly', () => {
    const { container } = render(<LoadingText />);
    expect(container).toMatchSnapshot();
  });

  it('ModalButtons renders correctly', () => {
    const { container } = render(<ModalButtons />);
    expect(container).toMatchSnapshot();
  });

  it('ModalCancellationButton renders correctly', () => {
    const { container } = render(<ModalCancellationButton />);
    expect(container).toMatchSnapshot();
  });

  it('ModalConfirmationButton renders correctly', () => {
    const { container } = render(<ModalConfirmationButton />);
    expect(container).toMatchSnapshot();
  });

  it('ModalContent renders correctly', () => {
    const { container } = render(<ModalContent />);
    expect(container).toMatchSnapshot();
  });

  it('ModalError renders correctly', () => {
    const { container } = render(<ModalError />);
    expect(container).toMatchSnapshot();
  });

  it('ModalInput renders correctly', () => {
    const { container } = render(<ModalInput />);
    expect(container).toMatchSnapshot();
  });

  it('ModalText renders correctly', () => {
    const { container } = render(<ModalText />);
    expect(container).toMatchSnapshot();
  });

  it('ModalTitle renders correctly', () => {
    const { container } = render(<ModalTitle />);
    expect(container).toMatchSnapshot();
  });

  it('RedIcon renders correctly', () => {
    const { container } = render(<RedIcon />);
    expect(container).toMatchSnapshot();
  });
});
