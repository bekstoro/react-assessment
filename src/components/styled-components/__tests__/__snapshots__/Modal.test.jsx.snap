// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal styled components BaseModal renders correctly 1`] = `
.c0 {
  position: absolute;
  background-color: #fff;
  border: 2px solid #ea2929;
  border-radius: 15px;
  padding: 30px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  z-index: 100;
}

<div>
  <div
    class="c0"
  />
</div>
`;

exports[`Modal styled components BigModal renders correctly 1`] = `
.c0 {
  position: absolute;
  background-color: #fff;
  border: 2px solid #ea2929;
  border-radius: 15px;
  padding: 30px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  z-index: 100;
}

.c1 {
  width: 600px;
  height: 300px;
  top: calc(50% - 150px);
  left: calc(50% - 330px);
}

<div>
  <div
    class="c0 c1"
  />
</div>
`;

exports[`Modal styled components LoadingIcon renders correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  border: 10px solid #174d73;
  border-top: 10px solid transparent;
  border-radius: 50%;
  width: 60px;
  height: 60px;
  text-align: center;
  -webkit-animation-name: spin;
  animation-name: spin;
  -webkit-animation-duration: 2s;
  animation-duration: 2s;
  -webkit-transition-timing-function: linear;
  transition-timing-function: linear;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
  -webkit-animation: eoUyJr 2s linear infinite;
  animation: eoUyJr 2s linear infinite;
}

<div>
  <div
    class="c0"
  />
</div>
`;

exports[`Modal styled components LoadingText renders correctly 1`] = `
.c0 {
  font-weight: 500;
  color: #3498db;
  margin-bottom: 0;
}

<div>
  <h3
    class="c0"
  />
</div>
`;

exports[`Modal styled components ModalButtons renders correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  width: 100%;
}

<div>
  <section
    class="c0"
  />
</div>
`;

exports[`Modal styled components ModalCancellationButton renders correctly 1`] = `
.c0 {
  width: 100px;
  padding: 10px 0;
  background-color: #fff;
  border: 1px solid #03629a;
  border-radius: 4px;
  color: #03629a;
  font-weight: bold;
  margin-left: 10px;
  cursor: pointer;
}

.c0:hover {
  opacity: 0.5;
}

<div>
  <button
    class="c0"
  />
</div>
`;

exports[`Modal styled components ModalConfirmationButton renders correctly 1`] = `
.c0 {
  width: 100px;
  padding: 10px 0;
  background-color: #d7eef7;
  border: 1px solid #d7eef7;
  border-radius: 4px;
  color: #03629a;
  font-weight: bold;
  cursor: pointer;
}

.c0:hover {
  opacity: 0.5;
}

<div>
  <button
    class="c0"
  />
</div>
`;

exports[`Modal styled components ModalContent renders correctly 1`] = `
.c0 {
  width: 100%;
}

.c0 {
  width: 100%;
  text-align: left;
}

.c0 textarea {
  width: 100%;
  height: 80px;
  resize: none;
  color: #03629a;
}

<div>
  <section
    class="c0"
  />
</div>
`;

exports[`Modal styled components ModalError renders correctly 1`] = `
.c0 {
  color: #ea2929;
  margin: 0;
}

<div>
  <h5
    class="c0"
  />
</div>
`;

exports[`Modal styled components ModalInput renders correctly 1`] = `
.c0 {
  width: 200px;
}

<div>
  <input
    class="c0"
  />
</div>
`;

exports[`Modal styled components ModalText renders correctly 1`] = `
.c0 {
  font-weight: normal;
  font-size: 0.9rem;
  margin: 0;
}

<div>
  <h4
    class="c0"
  />
</div>
`;

exports[`Modal styled components ModalTitle renders correctly 1`] = `
.c0 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 1.2rem;
  margin-top: 0;
  padding-bottom: 15px;
  border-bottom: 1px solid #c1c1c1;
}

<div>
  <h3
    class="c0"
  />
</div>
`;

exports[`Modal styled components RedIcon renders correctly 1`] = `
.c0 {
  color: #ea2929;
  margin-right: 10px;
}

<div>
  <i
    class="c0"
  />
</div>
`;

exports[`Modal styled components SmallModal renders correctly 1`] = `
.c0 {
  position: absolute;
  background-color: #fff;
  border: 2px solid #ea2929;
  border-radius: 15px;
  padding: 30px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  z-index: 100;
}

.c1 {
  width: 600px;
  height: 130px;
  top: calc(50% - 65px);
  left: calc(50% - 330px);
}

<div>
  <div
    class="c0 c1"
  />
</div>
`;

exports[`Modal styled components SmallModalCentered renders correctly 1`] = `
.c0 {
  position: absolute;
  background-color: #fff;
  border: 2px solid #ea2929;
  border-radius: 15px;
  padding: 30px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  z-index: 100;
}

.c1 {
  width: 600px;
  height: 130px;
  top: calc(50% - 65px);
  left: calc(50% - 330px);
}

.c2 {
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<div>
  <div
    class="c0 c1 c2"
  />
</div>
`;
